### PR TITLE
Readme fixup part 1

### PR DESCRIPTION
**Pull request by JoelEinbinder**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -135,19 +135,19 @@ The following is a typical example of using a Puppeteer to drive automation:
...
   // other actions...
   browser.close();
 });
 ```
 
 #### puppeteer.connect(options)
-- `options` <[Object]>  Set of options to connect to the browser. Can have the following fields:
-  - `remoteDebuggingURL` <[string]> a remote debugging URL to connect to.
+- `options` <[Object]>
+  - `remoteDebuggingURL` <[string]> a [remote debugging URL](#browserremotedebuggingurl) to connect to.
```
Comment (paulirish) [inline docs/api.md:146]: fwiw this will conflict with #272

Comment (JoelEinbinder) [reply docs/api.md:146]: This patch will conflict with everything. It touches the api.md

File: docs/api.md [hunk 2]
```diff
@@ -245,42 +242,42 @@ Emitted when the page crashes.
...
 
-Emitted when a page's `load` event was dispatched.
+Emitted when the JavaScriot [`load`](https://developer.mozilla.org/en-US/docs/Web/Events/load) event is dispatched.
 
 #### event: 'pageerror'
-- <[string]>
+- <[string]> The exception message
 
-Emitted when an unhandled exception happens on the page. The only argument of the event holds the exception message.
+Emitted when an unhandled exception happens on the page.
```
Comment (paulirish) [inline docs/api.md]: This is a pedantic nit, but... `unhandled` -> `uncaught`, right? 

Separately, do you get this event on an unhandled promise rejection?

File: docs/api.md [hunk 3]
```diff
@@ -289,78 +286,17 @@ Emitted when a [response] is received.
 
 #### page.$(selector)
 - `selector` <[string]> Selector to query page for
-- returns: <[Promise]<[ElementHandle]>> Promise which resolves to ElementHandle pointing to the page element.
+- returns: <[Promise]<[ElementHandle]>>
 
-The method queries page for the selector. If there's no such element on the page, the method will resolve to `null`.
+The method runs `document.querySelector` on the page. If no element matches the selector, the return value resolve to `null`.
```
Comment (paulirish) [inline docs/api.md]: "on the page" => "within the page"

File: docs/api.md [hunk 4]
```diff
@@ -289,78 +286,17 @@ Emitted when a [response] is received.
...
-    console.log(content);
-  });
-  browser.close();
-});
-
-```
-
 #### page.addScriptTag(url)
-- `url` <[string]> Url of a script to be added
+- `url` <[string]> Sets the url of the `<script>` tag
```
Comment (paulirish) [inline docs/api.md]: i'd nuke "sets the".

File: docs/api.md [hunk 5]
```diff
@@ -289,78 +286,17 @@ Emitted when a [response] is received.
...
 - returns: <[Promise]> Promise which resolves as the script gets added and loads.
 
-Adds a `<script>` tag to the frame with the desired url. Alternatively, JavaScript could be injected to the frame via [`frame.injectFile`](#frameinjectfilefilepath) method.
+Adds a `<script>` tag into the page with the desired url. Alternatively, JavaScript could be injected to the page via [`page.injectFile`](#pageinjectfilefilepath) method.
```
Comment (paulirish) [inline docs/api.md]: " Alternatively, JavaScript could be injected" => " Alternatively, a local JavaScript file could be injected"

File: docs/api.md [hunk 6]
```diff
@@ -289,78 +286,17 @@ Emitted when a [response] is received.
...
-  });
-  browser.close();
-});
-
-```
-
 #### page.addScriptTag(url)
-- `url` <[string]> Url of a script to be added
+- `url` <[string]> Sets the url of the `<script>` tag
 - returns: <[Promise]> Promise which resolves as the script gets added and loads.
```
Comment (paulirish) [inline docs/api.md]: > which resolves when the script's `onload` fires.

File: docs/api.md (hunk 5 above)
Comment (paulirish) [reply docs/api.md]: side note: we don't reject onerror, so a bad URL will just hang forever.

File: docs/api.md [hunk 7]
```diff
@@ -371,25 +307,31 @@ Shortcut for [page.mainFrame().addScriptTag(url)](#frameaddscripttagurl).
   - `button` <[string]> `left`, `right`, or `middle`, defaults to `left`.
   - `clickCount` <[number]> defaults to 1
```
Comment (paulirish) [inline docs/api.md]: Wanna explain clickCount here?

File: docs/api.md [hunk 8]
```diff
@@ -371,25 +307,31 @@ Shortcut for [page.mainFrame().addScriptTag(url)](#frameaddscripttagurl).
...
-- returns: <[Promise]> Returns promise which resolves when page gets closed.
+- returns: <[Promise]>
 
 #### page.emulate(options)
 - `options` <[Object]>
-  - `viewport` <[Object]> viewport as described in [`page.setViewport`](#pagesetviewportpath) method.
-  - `userAgent` <[string]> user agent string
-- returns: <[Promise]> Promise which resolves when emulation is performed.
+  - `viewport` <[Object]>
+    - `width` <[number]> Specifies the page width in pixels.
```
Comment (paulirish) [inline docs/api.md]: similar to the "sets the " i think the docs of args & options shouldn't have "Specifies the". Right now on master only `uploadFile` it the only other place we have this sort of wording.

File: docs/api.md [hunk 9]
```diff
@@ -435,18 +377,78 @@ Shortcut for [page.mainFrame().evaluate(pageFunction, ...args)](#frameevaluatepa
...
-- whenever the page gets navigated
-- whenever the child frame gets attached or navigated. In this case, the function gets invoked in the context of the newly attached frame
+- whenever the page is navigated
+- whenever the child frame is attached or navigated. In this case, the function is invoked in the context of the newly attached frame
 
 The function is invoked after the document was created but before any of its scripts were run. This is useful to amend JavaScript environment, e.g. to seed [Math.random](https://github.com/GoogleChrome/puppeteer/blob/master/examples/unrandomize.js)
 
+#### page.exposeFunction(name, puppeteerFunction)
+- `name` <[string]> Name of the function on the window object
+- `puppeteerFunction` <[function]> Callback function which will be called in Puppeteer's context.
```
Comment (paulirish) [inline docs/api.md:391]: I don't love this name, but the alternatives I'm thinking aren't compelling enough to change. :expressionless:

File: docs/api.md [hunk 10]
```diff
@@ -435,18 +377,78 @@ Shortcut for [page.mainFrame().evaluate(pageFunction, ...args)](#frameevaluatepa
...
+- returns: <[Promise]> Promise which resolves with the result of `puppeteerFunction`.
+
+The method adds a function called `name` on the `window` object.
```
Comment (paulirish) [inline docs/api.md]: > on the page's window object.

File: docs/api.md [hunk 11]
```diff
@@ -435,18 +377,78 @@ Shortcut for [page.mainFrame().evaluate(pageFunction, ...args)](#frameevaluatepa
...
+  });
+  browser.close();
+});
+
+```
+
 #### page.focus(selector)
-- `selector` <[string]> A query [selector] of element to focus. If there are multiple elements satisfying the selector, the first will be focused.
-- returns: <[Promise]> Promise which resolves when the element matching `selector` is successfully focused. Promise gets rejected if there's no element matching `selector`.
+- `selector` <[string]> A query [selector] of an element to focus. If there are multiple elements satisfying the selector, the first will be focused.
```
Comment (paulirish) [inline docs/api.md]: > A [selector] of an element to focus.

selectors aren't called "query selectors", afaik

File: docs/api.md [hunk 12]
```diff
@@ -456,7 +458,7 @@ Shortcut for [page.mainFrame().focus(selector)](#framefocusselector).
 #### page.goBack(options)
 - `options` <[Object]> Navigation parameters which might have the following properties:
   - `timeout` <[number]> Maximum navigation time in milliseconds, defaults to 30 seconds.
-  - `waitUntil` <[string]> When to consider navigation succeeded, defaults to `load`. Could be either:
+  - `waitUntil` <[string]> When to consider a navigation finished, defaults to `load`. Could be either:
     - `load` - consider navigation to be finished when the `load` event is fired.
     - `networkidle` - consider navigation to be finished when the network activity stays "idle" for at least `networkIdleTimeout`ms.
```
Comment (paulirish) [inline docs/api.md]: add space before `ms`

File: docs/api.md [hunk 13]
```diff
@@ -740,13 +745,13 @@ Shortcut for [page.mainFrame().waitForFunction(pageFunction[, options, ...args])
...
     - `load` - consider navigation to be finished when the `load` event is fired.
-    - `networkidle` - consider navigation to be finished when the network activity stays "idle" for at least `networkIdleTimeout`ms.
+    - `networkidle` - consider navigation to be finished when the network activity stays "idle" for at least `networkIdleTimeout` ms.
   - `networkIdleInflight` <[number]> Maximum amount of inflight requests which are considered "idle". Takes effect only with `waitUntil: 'networkidle'` parameter.
   - `networkIdleTimeout` <[number]> A timeout to wait before completing navigation. Takes effect only with `waitUntil: 'networkidle'` parameter.
 - returns: <[Promise]<[Response]>> Promise which resolves to the main resource response. In case of multiple redirects, the navigation will resolve with the response of the last redirect.
 
 #### page.waitForSelector(selector[, options])
-- `selector` <[string]> CSS selector of awaited element,
+- `selector` <[string]> A query [selector] of an element to wait for,
```
Comment (paulirish) [inline docs/api.md]: > A [selector]

Agree it's not a "CSS selector". https://www.w3.org/TR/selectors4/

File: docs/api.md [hunk 14]
```diff
@@ -1054,7 +1059,7 @@ puppeteer.launch().then(async browser => {
 ```
 
 #### frame.waitForSelector(selector[, options])
-- `selector` <[string]> CSS selector of awaited element,
+- `selector` <[string]> A query [selector] of an element to wait for,
```
Comment (paulirish) [inline docs/api.md]: > A [selector]

**Review by paulirish — CHANGES_REQUESTED** [review]
really like the wording and clarity changes here.
also removing unnecessary words. so good.

File: docs/api.md (hunk 2 above)
Comment (JoelEinbinder) [reply docs/api.md]: I'm not sure.

Comment (JoelEinbinder) [reply docs/api.md]: Yes, it gets unhandled promise rejection.

**Comment by aslushnikov** [comment]
@JoelEinbinder do you plan to land this in the nearest future? If yes, I'd wait with all the other patches that change api.md

**Comment by JoelEinbinder** [comment]
I am fine with rebasing it. But I was planning to land it at night when no patches are in the air.

**Review by paulirish — APPROVED** [review]
thx. lgtm

File: docs/api.md [hunk 15]
```diff
@@ -135,19 +135,19 @@ The following is a typical example of using a Puppeteer to drive automation:
...
-- `options` <[Object]>  Set of options to connect to the browser. Can have the following fields:
-  - `remoteDebuggingURL` <[string]> a remote debugging URL to connect to.
+- `options` <[Object]>
+  - `remoteDebuggingURL` <[string]> a [remote debugging URL](#browserremotedebuggingurl) to connect to.
   - `ignoreHTTPSErrors` <[boolean]> Whether to ignore HTTPS errors during navigation. Defaults to `false`.
-- returns: <[Promise]<[Browser]>> Promise which resolves to browser instance.
+- returns: <[Promise]<[Browser]>>
 
-This method could be used to connect to already running browser instance.
+This methods attaches Puppeteer to an existing Chromium instance.
```
Comment (aslushnikov) [inline docs/api.md:150]: nit: method

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**